src/legacy/core_plugins/kibana/public/management/sections/objects/components/objects_table/components/table/table.js

```diff
@@ -169,50 +167,60 @@ class TableUI extends PureComponent {
             id: 'kbn.management.objects.objectsTable.table.columnTypeDescription', defaultMessage: 'Type of the saved object'
           }),
         sortable: false,
-        render: type => {
+        render: (type, object) => {
           return (
             <EuiToolTip
               position="top"
               content={getSavedObjectLabel(type)}
             >
               <EuiIcon
                 aria-label={getSavedObjectLabel(type)}
-                type={getSavedObjectIcon(type)}
+                type={object.meta.icon || 'apps'}
                 size="s"
               />
             </EuiToolTip>
           );
         },
       },
       {
-        field: 'title',
+        field: 'meta.title',
         name: intl.formatMessage({ id: 'kbn.management.objects.objectsTable.table.columnTitleName', defaultMessage: 'Title' }),
         description:
         intl.formatMessage({
           id: 'kbn.management.objects.objectsTable.table.columnTitleDescription', defaultMessage: 'Title of the saved object'
         }),
         dataType: 'string',
         sortable: false,
-        render: (title, object) => (
-          <EuiLink href={getEditUrl(object.id, object.type)}>{title}</EuiLink>
-        ),
+        render: (title, object) => {
+          const { path } = object.meta.inAppUrl;
+          const canGoInApp = this.props.canGoInApp(object);
+          if (!canGoInApp) {
+            return (
+              <EuiText size="s">{title || getDefaultTitle(object)}</EuiText>
+            );
+          }
+          return (
+            <EuiLink href={chrome.addBasePath(path)}>{title || getDefaultTitle(object)}</EuiLink>
+          );
+        },
       },
       {
         name: intl.formatMessage({ id: 'kbn.management.objects.objectsTable.table.columnActionsName', defaultMessage: 'Actions' }),
         actions: [
           {
             name: intl.formatMessage({
-              id: 'kbn.management.objects.objectsTable.table.columnActions.viewInAppActionName', defaultMessage: 'In app'
+              id: 'kbn.management.objects.objectsTable.table.columnActions.editActionName',
+              defaultMessage: 'Edit'
             }),
             description:
               intl.formatMessage({
-                id: 'kbn.management.objects.objectsTable.table.columnActions.viewInAppActionDescription',
-                defaultMessage: 'View this saved object within Kibana'
+                id: 'kbn.management.objects.objectsTable.table.columnActions.editActionDescription',
+                defaultMessage: 'Edit this saved object'
```

Comment:
nit: We're displaying an "Edit this saved object" tooltip even when the user only has "read" access.

Reply:
Changed it to `inspect` terminology.